sky/sdk/example/game/main.dart

```diff
@@ -90,13 +91,97 @@ class GameDemoApp extends App {
 
   Widget _buildMainScene(navigator, route) {
     return new Center(
-      child: new RaisedButton(
-        child: new Text("Play"),
+      child: new TextureButton(
         onPressed: () {
           _game = new GameDemoWorld(_app, navigator, _loader, _spriteSheet, _spriteSheetUI);
           navigator.pushNamed('/game');
-        }
+        },
+        texture: _spriteSheetUI["btn_play_up.png"],
+        textureDown: _spriteSheetUI["btn_play_down.png"],
+        width: 128.0,
+        height: 128.0
       )
     );
   }
 }
+
+class TextureButton extends ButtonBase {
+  TextureButton({
+    Key key,
+    this.onPressed,
+    this.texture,
+    this.textureDown,
+    this.width: 128.0,
+    this.height: 128.0
+  }) : super(key: key);
+
+  final Function onPressed;
+  final Texture texture;
+  final Texture textureDown;
+  final double width;
+  final double height;
+
+  Widget buildContent() {
+    return new Listener(
+      child: new Container(
+        width: width,
+        height: height,
+        child: new CustomPaint(
+          callback: paintCallback,
+          token: new _TextureButtonToken(
+            highlight,
+            texture,
+            textureDown,
+            width,
+            height
+          )
+        )
+      ),
+      onPointerUp: (_) {
+        if (onPressed != null) onPressed();
+      }
+    );
+  }
+
+  void paintCallback(PaintingCanvas canvas, Size size) {
+    if (texture == null)
+      return;
+
+    if (highlight && textureDown != null) {
+      // Draw down state
+      canvas.scale(size.width / textureDown.size.width, size.height / textureDown.size.height);
+      textureDown.drawTexture(canvas, Point.origin, new Paint());
+    } else {
+      // Draw up state
+      canvas.scale(size.width / texture.size.width, size.height / texture.size.height);
+      texture.drawTexture(canvas, Point.origin, new Paint());
+    }
+  }
+}
+
+class _TextureButtonToken {
+  _TextureButtonToken(
+    this._highlight,
+    this._texture,
+    this._textureDown,
+    this._width,
+    this._height
+  );
+
+  final bool _highlight;
+  final Texture _texture;
+  final Texture _textureDown;
+  final double _width;
+  final double _height;
+
+  bool operator== (other) {
+    return (
+      other is _TextureButtonToken &&
+      _highlight == other._highlight &&
+      _texture == other._texture &&
+      _textureDown == other._textureDown &&
+      _width == other._width &&
+      _height == other._height
+    );
+  }
+}
```

Comment:
you should implement `int get hashCode` on this class. Look around the codebase, there are examples of how to do it all over the place.